system_modules/napaudio/src/audio/utility/audiofileutils.cpp

```diff
@@ -159,12 +164,71 @@ namespace nap
 		
 		bool readAudioFile(const std::string& fileName, MultiSampleBuffer& output, float& outSampleRate, nap::utility::ErrorState& errorState)
 		{
-			if (utility::toLower(utility::getFileExtension(fileName)) == "mp3")
-				return readMp3File(fileName, output, outSampleRate, errorState);
+			// if (utility::toLower(utility::getFileExtension(fileName)) == "mp3")
+			// 	return readMp3File(fileName, output, outSampleRate, errorState);
 			
 			return readLibSndFile(fileName, output, outSampleRate, errorState);
 		}
 		
 		
+		//this function should probably live in nap::math
+		bool isFloatEqual(const float & a, const float & b) {
```

Comment:
a copy is preferred; it's definitely just as fast, if not faster than a reference. Passing a single float by value is smaller (4 bytes) than a pointer (8 bytes) on most 64 bit systems. Also; easier to optimize.

Reply:
Wouldn't this be faster and simpler anyway: `abs(a-b) > epsilon<float>()`, you want the absolute difference to be higher than min allowed difference.